File: Delete middle element of a stack[15-July-2023]/Solution.java
// >> JAVA CODE <<

class Solution
{
    //Function to delete middle element of a stack.
    private void solve(Stack<Integer> s, int sizeOfStack, int midInd) {
        if (sizeOfStack == 0) {
            return;
        }
        
        int el = s.pop();
        solve(s, sizeOfStack - 1, midInd);
        
        if (midInd != sizeOfStack) {
            s.add(el);
        }
    }
    public void deleteMid(Stack<Integer>s,int sizeOfStack){
        solve(s,sizeOfStack,(sizeOfStack+1)/2);
    } 
}

File: Delete middle element of a stack[15-July-2023]/Solution.py
#  >> PYTHON CODE <<

class Solution:
    #Function to delete middle element of a stack.
    def deleteMid(self, s, sizeOfStack):
        def solve(s,sizeOfStack,midInd):
            if(sizeOfStack==0): return
        
            el=s.pop()
            solve(s,sizeOfStack-1,midInd)
            
            if(midInd!=sizeOfStack):
                s.append(el)
        
        solve(s,sizeOfStack,(sizeOfStack+1)//2)
